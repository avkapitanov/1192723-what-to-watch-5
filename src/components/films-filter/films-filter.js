import React from "react";
import {ActionCreator} from "../../store/action";
import {connect} from "react-redux";
import PropTypes from "prop-types";

const FilmsFilter = (props) => {
  const {filterGenres, selectedFilterGenre, changeFilterGenre} = props;

  return (
    <ul className="catalog__genres-list">
      {
        filterGenres.map((genre) =>
          <li key={genre} className={`catalog__genres-item ${selectedFilterGenre === genre ? `catalog__genres-item--active` : ``}`}>
            <a href="#" className="catalog__genres-link" data-genre={genre} onClick={changeFilterGenre}>{genre}</a>
          </li>
        )
      }
    </ul>
  );
};

export {FilmsFilter};

FilmsFilter.propTypes = {
  changeFilterGenre: PropTypes.func.isRequired,
  selectedFilterGenre: PropTypes.string.isRequired,
  filterGenres: PropTypes.array.isRequired
};

const mapStateToProps = ({DATA, PROCESS}) => ({
  selectedFilterGenre: PROCESS.selectedFilterGenre,
  filterGenres: DATA.filterGenres
});

const mapDispatchToProps = (dispatch) => ({
  changeFilterGenre(evt) {
    evt.preventDefault();
    dispatch(ActionCreator.changeFilterGenre(evt.target.dataset.genre));
    dispatch(ActionCreator.resetRenderedFilmsCount());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(FilmsFilter);
