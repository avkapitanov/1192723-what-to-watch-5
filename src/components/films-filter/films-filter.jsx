import React from "react";
import {changeFilterGenre} from "../../store/action";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {getFilterGenres, getSelectedGenre} from "../../store/selectors";

const FilmsFilter = (props) => {
  const {filterGenres, selectedFilterGenre, changeFilterGenreAction} = props;

  return (
    <ul className="catalog__genres-list">
      {
        filterGenres.map((genre) =>
          <li key={genre} className={`catalog__genres-item ${selectedFilterGenre === genre ? `catalog__genres-item--active` : ``}`}>
            <a href="#" className="catalog__genres-link" data-genre={genre} onClick={changeFilterGenreAction}>{genre}</a>
          </li>
        )
      }
    </ul>
  );
};

export {FilmsFilter};

FilmsFilter.propTypes = {
  changeFilterGenreAction: PropTypes.func.isRequired,
  selectedFilterGenre: PropTypes.string.isRequired,
  filterGenres: PropTypes.array.isRequired
};

const mapStateToProps = (state) => ({
  selectedFilterGenre: getSelectedGenre(state),
  filterGenres: getFilterGenres(state)
});

const mapDispatchToProps = (dispatch) => ({
  changeFilterGenreAction(evt) {
    evt.preventDefault();
    dispatch(changeFilterGenre(evt.target.dataset.genre));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(FilmsFilter);
