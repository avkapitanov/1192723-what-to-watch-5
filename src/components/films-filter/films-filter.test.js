import React from "react";
import renderer from "react-test-renderer";
import {FilmsFilter} from "./films-filter";

const filterGenres = [
  `All genres`,
  `Drama`,
  `Comedy`,
  `Horror`,
  `Romance`,
  `Sci-Fi`
];

const selectedFilterGenre = `Drama`;

it(`FilmsFilter component render correctly`, () => {
  const tree = renderer.create(
      <FilmsFilter
        filterGenres={filterGenres}
        selectedFilterGenre={selectedFilterGenre}
        onChangeFilterGenre={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
