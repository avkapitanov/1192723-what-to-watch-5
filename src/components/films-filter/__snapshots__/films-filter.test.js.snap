// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsFilter component render correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="All genres"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      data-genre="Drama"
      href="#"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Comedy"
      href="#"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Horror"
      href="#"
      onClick={[Function]}
    >
      Horror
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Romance"
      href="#"
      onClick={[Function]}
    >
      Romance
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      data-genre="Sci-Fi"
      href="#"
      onClick={[Function]}
    >
      Sci-Fi
    </a>
  </li>
</ul>
`;
